libpod/container_internal_linux.go

```diff
@@ -1942,9 +1942,24 @@ func (c *Container) generateHosts(path string) (string, error) {
 	}
 	hosts := string(orig)
 	hosts += c.getHosts()
+
+	hosts = c.appendLocalhost(hosts)
+
 	return c.writeStringToRundir("hosts", hosts)
 }
 
+// based on networking mode we may want to append the localhost
+// if there isn't any record for it and also this shoud happen
+// in slirp4netns and similar network modes.
+func (c *Container) appendLocalhost(hosts string) string {
+	if !strings.Contains(hosts, "localhost") &&
+		c.config.NetMode.IsSlirp4netns() {
```

Comment:
I think `!c.config.NetMode.IsHost()` is better. Only for host we don't want to add this.

Reply:
Thanks for mentioning this, I have update code and tests.